Mutantrace specflag omnibus

Incoming5643 opened the pull request:
EASYDISMEMBER races (skeletons and zombies) have over double the chance to get dismembered when attacked with something that can dismember. For example a 20% chance weapon goes from 20% to an effective 52%

EASYLIMBATTACHMENT allows races (skeletons and zombies) with the flag to just shove limbs found lying on the floor into empty limb slots to regain their use. However this DOESN'T heal the damaged limbs, so in most cases doing this will actually add damage to the mob. Comically, the limbs attached don't need to be the originals, so these races can go around ripping off other peoples limbs and then using them for themselves.

TOXINLOVER races (Slime and Jelly people, as well as zombies) take inverse toxin damage. Things that typically heal toxin damage will instead cause it and visa versa. Note that the additional effects of the toxins/medicines still work as normal. Additionally this ties the mutant's blood level to these transactions, reaffirming the slime and jelly peoples vulnerability to antitoxin and the like, which had been lost in the blood rewrite. **I'm counting this as the fix for this PR**

---

New ability/disabilities for slime/jelly people:

Regenerate limbs, that allows these races to spend jelly to reform lost limbs

Slime/Jelly people also gain a new "power" that forces their limbs to be eaten for jelly if the slimeperson is in threat of dying from a lack of jelly.

Slime/Jelly people can now be dismembered

---

Pod people now mutate if shot by the floral somatoray.

:cl: incoming5643
experiment: New abilities for some races!
rscadd: Skeletons (liches) and zombies now are much more susceptible to limb loss, but can also reattach lost limbs manually without surgery.
rscadd: Slimepeople can now lose their limbs, but also can regenerate them if they have the jelly for it. Additionally slimepeople will (involuntarily) lose limbs and convert them to slime jelly if they're extremely low on jelly.
rscadd: Zombies and slimepeople now have reversed toxic damage, what usually heals now hurts and what usually hurts now heals. Keep in mind that while you could for example heal with unstable mutagen, you'd still mutate. Beware of secondary effects! Drugs like antitoxin are especially dangerous to these races now since not only do they do toxin damage, but they remove other toxic chems before they can heal you. Be wary also of wide band healing drugs like omnizine.
rscadd: For slime people the above mechanic also adds/drains slime jelly. So yes inhaling plasma is now a valid way to build jelly (but you'll still suffocate if there's no oxygen).
rscadd: Pod people now mutate if shot with the mutation setting of the floral somatoray.
/:cl:

Incoming5643 committed:
Adds the EASYDISMEMBER and EASYLIMBATTACHMENT specflags

EASYDISMEMBER races have about double the chance to get dismembered when attacked with something that can dismember.

EASYLIMBATTACHMENT allows races with the flag to just shove limbs found lying on the floor into empty limb slots to regain their use. However this DOESN'T heal the damaged limbs, so in most cases doing this will actually add damage to the mob. Comically, the limbs attached don't need to be the originals, so these races can go around ripping off other peoples limbs and then using them for themselves.

Both these abilities are given to the the zombie and skeleton races, and constitute a significant nerf to lichs

coiax commented:
Only high functioning zombies are going to be able to use ripped limbs, given infectious zombies can't use their hands for anything other than clawing things.

Cebasiz commented:
blunt damage should do extra damage to skeletons tbh

Incoming5643 commented:
@coiax if you're feeling generous/suicidal you can snap new limbs onto these races for them, even if they can't manage it for themselves.

phil235 commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -1107,7 +1107,8 @@
 	//dismemberment
-	if(prob(I.get_dismemberment_chance(affecting)))
+	var/probability = I.get_dismemberment_chance(affecting)
+	if(prob(probability) || (EASYDISMEMBER in specflags && prob(2*probability)))
```
(EASYDISMEMBER in specflags) &&

Incoming5643 replied:
seemed to work right in testing but you're right `in` is really bad about that

phil235 replied:
better safe than sorry

phil235 commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -45,6 +45,21 @@
+/obj/item/bodypart/attack(mob/living/carbon/C, mob/user)
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.dna && H.dna.species && EASYLIMBATTACHMENT in H.dna.species.specflags)
```
just an fyi, you don't need to check for dna and dna.species, humans always have those. The only moment where a human doesn't have them is when he's being qdel'd.

ChangelingRain commented on code/modules/surgery/bodyparts/bodyparts.dm:
```diff
@@ -45,6 +45,21 @@
+					"<span class='notice'>You force [src] into your empty socket, and it locks into place!</span>")
+				else
+					H.visible_message("<span class='warning'>[user] jams [src] into [H]'s empty socket!</span>",\
+					"<span class='notice'>[user] force [src] into your empty socket, and it locks into place!</span>")
```
forces

Zilenan91 commented:
I thought dismemberment was damage based not percentage based

Supermichael777 commented:
It is percent based and that percentage scales with damage.

Incoming5643 commented:
It's both, to even roll for dismemberment a limb needs to have taken some damage

`get_damage() >= (max_damage - I.armour_penetration/2`

but said roll is dependent on the magnatude of the damage and the weight class of the item doing the current damage

`if((sharpness || damtype == BURN) && w_class >= 3)`
			`. = force*(w_class-1)`

Razharas commented:
@Incoming5643 
Your probability is not just doubled thought, since you first check usual dismemberment prob regardless of flags and then if theres a flag you roll again your final probability to dismember is:
````
(probability + ((1 - probability) * probability * 2)) * 100%
````

Thus 10% chance to dismember will result in 28% chance to dismembet with the flag, not 20%

RemieRichards commented:
@Razharas read his OP:

>For example a 20% chance weapon goes from 20% to an effective 48%

Razharas commented:
20% should result in 52% though
1 - 0.2 = 0.8
0.2 * 2 = 0.4
0.8 * 0.4 = 0.32
0.32 + 0.2 = 0.52

(1 - 0.2) * (0.2 * 2) + 0.2

RemieRichards commented:
Ye, but the point is he knows it's not "just doubling"

Razharas commented:
Well i didnt read the op post

Steelpoint commented:
How does this nerf lichs?

Unless removing a lichs head prevents them from respawning then this change does almost nothing to nerf the power of a lich.

Incoming5643 commented:
It gives the crew an avenue to cripple the lich without killing him, making him less effective over his lifespan even if it can still resurrect.

I'm not removing the resurrection aspect, that's literally the whole point of the spell.

Steelpoint commented:
Does losing a limb carry over to a new body?

Incoming5643 commented:
Not currently, no.

Steelpoint commented:
Then this is worthless and does almost nothing to nerf the lich.

Why would players risk keeping a Lich alive but disarmed of their limbs when even then the Lich can still use their spells unabated? Not to mention the lich can just commit suicide to trigger the regeneration.

Once again, if someone's in a position to cut off a lich's limbs then the lich is already dead irrespective. 

This is a nice change, but it has no negative effect on lichs whatsoever as the negative trait is so minor.

Incoming5643 commented:
Fighting a (good) wizard is usually a battle of attrition anyways, so if a random assistant who otherwise couldn't do much to the wizard could manage to lop off a limb with a thrown spear then run off with it it does more over the long term of "the fight" than most other potential changes.

Sure they can kill themselves to regain their limbs, but that means intentionally opening themselves up to discovery and a lengthy vulnerability period. It's a risk/reward consideration, which I thought was nice.

Steelpoint commented:
Its nice but not a big enough nerf to lich, on the set of scales the reward of having multiple lives after death is a very, very, heavy weight whereas having a higher chance to lose a limb is simply not enough to even start tipping the scales in my opinion.

I must stress that being outright immune to bombs and one off traps as a sole Wizard is a massive advantage, what use is a trap when the Lich resurrects from it and now is aware of its exsistance and is now useless?

Steelpoint commented:
I should note that by itself this is a positive change.

The reason I'm annoyed is that this change by itself is not a sufficient enough nerf to the lich to have the revert PR closed.

Incoming5643 commented:
>immune to bombs

If the liches' body gets gibbed it cant come back. That's always been the case.

Steelpoint commented:
Except your bombing has to be on point, if you fail to gib then ggnore you've fucked up and the lich is now laughing.

Made worse if you get stunned/shot at so you have the choice to either die prematuraly or detonate and just kill the lich, not gib.

Incoming5643 commented:
The wizard robes offer pretty good resistance to bombs to begin with, don't quote me on this but I'm pretty sure a full health wizard could survive a bomb outside of its heavy destruction range even if they weren't a lich. Provided they didn't get embedded with a ton of shit.

Incoming5643 commented:
I mean the black lich robes have the same armoring as normal wizard robes, I could make them worse if you want if you want liches to go down easier/more frequently.

Steelpoint commented:
My ideal nerf would be Lich only getting one extra life.

Well actually my idea nerf would be removing it entirely since its such a broken spell. In the hands of a bad player its powerful, in the hands of a veteran player they're unstoppable. 

I disagree with just nerfing other items to compensate for the lich's power or making them subtypes of a subtype of a subtype.

Incoming5643 commented:
Well I mean there's a million ways to nerf it that aren't "removing the spell" or "undermining the key feature of the spell". I'm not against more nerfs I'm just trying to find some middle ground here.

Also nerfs are features now?

Incoming5643 commented:
For real now if this a feature I'm going to add a ton of specflags to this, some of which rely on not yet merged code. I'm not gonna waste a memetoken on a balance fix that doesn't even placate the de facto standard bearer of the fix

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into no_bones_about_it

Incoming5643 commented:
S C O P E  E X P A N S I O N

Incoming5643 committed:
Slime/Jelly people gain a new power, regenerate limbs, that allows them to spend jelly to reform lost limbs

Slime/Jelly people also gain a new "power" that forces their limbs to be eaten for jelly if the slimeperson is in threat of dying from a lack of jelly.

Slime/Jelly people can now be dismembered

New specflag: TOXINLOVER

Species with TOXINLOVER (Slime and Jelly people, as well as zombies) take inverse toxin damage, things that typically heal toxin damage will instead cause it and visa versa. Additionally this ties the mutant's blood level to these transactions, reaffirming the slime and jelly peoples vulnerability to antitoxin and the like, which had been lost in the blood rewrite

Dorsisdwarf commented:
Make lich respawn increase exponentially, and the phylactery shows up on GPS after the third/fourth rez

optimumtact commented:
Reduce scope! Fire the finishing torpedo

Incoming5643 commented:
I don't want to burn a token on anything else than the absolute maximum amount of content I can muster from it.

Learn the consequences of this silly system.


But I'll finish it before the weekend's out

optimumtact commented:
It's easy to acquire tokens man, dont' be silly

Incoming5643 committed:
Fixes poited out errors and spelling mistakes
the floral suppository now mutates pod people
adds a proper icon for limb regeneration
slime people can't consume their own head

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into no_bones_about_it

# Conflicts:
#	icons/mob/actions.dmi

Incoming5643 commented:
We should be good here

ChangelingRain commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -781,5 +781,12 @@
+			blood_volume -= 5*amount
+		else
+			blood_volume -= amount
+	..()
```
return ..()
(ideally all damage procs would return the damage done but not all of them do so yet)

ChangelingRain commented on code/modules/mob/living/carbon/carbon.dm:
```diff
@@ -781,5 +781,12 @@
 
-
+/mob/living/carbon/adjustToxLoss(amount, updating_health=1)
+	if(TOXINLOVER in dna.species.specflags) //damage becomes healing and healing becomes damage
```
Also I don't think carbon species have dna? I'm not sure, dna is weird.

Incoming5643 replied:
carbon species have var/datum/dna/dna = null
Humans have dna, and monkeys have dna
Aliens and "true demons" might not. I guess I should check all the same in this situation.

Incoming5643 committed:
Spotcheck

optimumtact commented:
@tgstation/commit-access

optimumtact commented:
Looks okay to me
